src/parser/chunkparser.cpp

```diff
@@ -0,0 +1,16 @@
+#include "chunkparser.hpp"
+
+namespace docgen {
+
+void ChunkParser::parse(const char *begin, 
+                        const char *end, 
+                        nlohmann::json& json)
+{
+    while (begin < end) {
```

Comment:
I kinda prefer `!=` stylistically (and I think is more common in C++ as well with iters like this), but also this literally doesn't matter at all and I'm just tryna nitpick so dw about it xD